### PR TITLE
appscale-tools 2.3.0

### PR DESCRIPTION
**Pull request by dunn**

### CODE REVIEW

**Comment by tdsmith** [comment]
Would you mind rewriting `install` to look like this? https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Python-for-Formula-Authors.md#example

This should use the setup_install_args syntax throughout and boto (and the other dependencies) should be installed to a different prefix so that executables shouldn't get linked into HOMEBREW_PREFIX.

Calling appscale-tools's setup.py might be easier than copying files "by hand".

**Comment by tdsmith** [comment]
See also: #38758

**Comment by dunn** [comment]
Yeah, I should've checked for existing PRs, sorry about that.

I think I figured it out, but I'm not familiar with the Python build process so let me know if it's wrong.

File: Library/Formula/appscale-tools.rb [hunk 1]
```diff
@@ -16,62 +16,65 @@ class AppscaleTools < Formula
...
     end
 
-    inreplace Dir["bin/appscale*"], /^  lib = os.*/, "  lib = '#{libexec}'"
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
+    system "python", *Language::Python.setup_install_args(libexec)
 
-    prefix.install "bin", "templates", "LICENSE", "README.md"
-    libexec.install Dir["lib/*"]
+    system "python", "./setup.py", "build"
+    system "python", "./setup.py", "install", "--prefix=#{libexec}"
```
Comment (tdsmith) [inline Library/Formula/appscale-tools.rb]: Install should imply build; does it complain if you do this in one step like the below?

`system "python", *Language::Python.setup_install_args(libexec)`

**Comment by tdsmith** [comment]
Looking good otherwise; thanks.

No worries re: other PRs; Jack caught me doing the same thing earlier.

File: Library/Formula/appscale-tools.rb (hunk 1 above)
Comment (dunn) [inline Library/Formula/appscale-tools.rb]: Oh, so these three lines:
```
system "python", *Language::Python.setup_install_args(libexec)
system "python", "./setup.py", "build"
system "python", "./setup.py", "install", "--prefix=#{libexec}"
```

can be replaced by just the first
```
system "python", *Language::Python.setup_install_args(libexec)
```
?

Comment (tdsmith) [inline Library/Formula/appscale-tools.rb]: Oops, didn't even see it there. Yep, just the one; setup_install_args provides arguments to python that invoke `setup.py install` with a wrapper shim and a bunch of flags.

**Comment by tdsmith** [comment]
The failure is because this is missing some recursive dependencies, like these: https://github.com/kiorky/SOAPpy/blob/develop/setup.py#L29-L31

**Comment by tdsmith** [comment]
If you're up for an adventure you can run this through https://github.com/tdsmith/homebrew-pypi-poet yourself or else I can do it later with this as a starting point.

**Comment by dunn** [comment]
I think the adventure went OK.

**Comment by xu-cheng** [comment]
Thanks @dunn

**Pull request closed without merging**